lib/omniauth/strategies/openid_connect.rb

```diff
@@ -173,6 +174,10 @@ def authorize_uri
 
         opts.merge!(options.extra_authorize_params) unless options.extra_authorize_params.empty?
 
+        options.allow_authorize_params.each do |key|
+          opts[key] = request.params[key.to_s] unless opts.key?(key)
```

Comment:
**Question**: Why is it necessary to check if the request param already exists, i.e. `unless opts.key?(key)` ?

I am currently looking at a use case in which the option `acr_values` is set by default and it is necessary to override the option `acr_values` dynamically during runtime. This is currently not possible because `opts.key?(:acr_values)` evaluates to `true` and hence `opts[key] = request.params[key.to_s]` is not executed. The same problem when some request params are set through `extra_auhorize_params`.

@yannvery I noticed that you added this line some time ago. Do you happen to remember why this was necessary?

**Background**: I am currently working on a use case regarding step-up authencation that can be implemented by passing new values to the request param `acr_values` whenever the user wants to access or perform a sensitive action.

/cc @bufferoverflow